### PR TITLE
Train imagenet example to desired accuracy

### PR DESCRIPTION
**Pull request by yuxihu**
We can train resnet50_v1 with top-1 accuracy 76.23% on 8 AWS p3.16xl instances. Below is the script to run the training job:

```
mpirun -np 64 --hostfile ~/8hosts --bind-to none --map-by slot -x NCCL_DEBUG=INFO -x NCCL_MIN_NRINGS=4 -x MXNET_USE_OPERATOR_TUNING=0 -mca pml ob1 -mca btl ^openib /home/ubuntu/.virtualenvs/mxnet/bin/python /home/ubuntu/horovod/examples/mxnet_imagenet_resnet50.py --use-rec --last-gamma --data-nthreads 4 --dtype float32 --batch-size 128 --lr 0.05 --lr-mode poly --warmup-epochs 5 --num-epochs 90 --model resnet50_v1 --rec-train /media/ramdisk/train-480px-q95.rec --rec-train-idx /media/ramdisk/train-480px-q95.idx --rec-val /media/ramdisk/val-480px-q95.rec --rec-val-idx /media/ramdisk/val-480px-q95.idx
```

### CODE REVIEW

File: examples/mxnet_imagenet_resnet50.py [hunk 1]
```diff
@@ -16,25 +16,25 @@
...
 from mxnet.io import DataBatch, DataIter
-from mxnet.util import makedirs
-from mxnet.gluon.model_zoo.vision import get_model
-import horovod.mxnet as hvd
+
 
 # Training settings
 parser = argparse.ArgumentParser(description='MXNet ImageNet Example',
                                  formatter_class=argparse.ArgumentDefaultsHelpFormatter)
-parser.add_argument('--benchmark', type=int, default='0',
```
Comment (apeforest) [inline examples/mxnet_imagenet_resnet50.py]: This argument is actually being used in imagenet example in MXNet: https://github.com/apache/incubator-mxnet/blob/ceabcaac77543d99246415b2fb2d8c973a830453/example/image-classification/common/data.py

File: examples/mxnet_imagenet_resnet50.py [hunk 2]
```diff
@@ -57,19 +57,24 @@
...
-parser.add_argument('--warmup-lr', type=float, default=0.001,
-                    help='starting warmup learning rate (default: 0.001)')
-parser.add_argument('--warmup-epochs', type=int, default=5,
-                    help='number of warmup epochs (default: 5)')
+parser.add_argument('--warmup-lr', type=float, default=0.0,
+                    help='starting warmup learning rate (default: 0.0)')
+parser.add_argument('--warmup-epochs', type=int, default=0,
+                    help='number of warmup epochs (default: 0)')
+parser.add_argument('--last-gamma', action='store_true', default=False,
+                    help='whether to init gamma of the last BN layer in each bottleneck to 0.')
```
Comment (apeforest) [inline examples/mxnet_imagenet_resnet50.py]: add (default: False)?

Comment (yuxihu) [reply examples/mxnet_imagenet_resnet50.py]: will add

File: examples/mxnet_imagenet_resnet50.py (hunk 1 above)
Comment (yuxihu) [reply examples/mxnet_imagenet_resnet50.py]: I was following gluoncv example. I replace this one with use-rec which I think does the same thing: controlling if to use synthetic data.

File: examples/mxnet_imagenet_resnet50.py [hunk 3]
```diff
@@ -83,32 +93,45 @@
...
-batch_size = args.batch_size
 num_classes = 1000
 num_training_samples = 1281167
-
-data_nthreads = args.data_nthreads
-context = mx.cpu() if args.no_cuda else mx.gpu(hvd.local_rank())
-
 batch_size = args.batch_size
+epoch_size = \
+    int(math.ceil(int(num_training_samples // num_workers) / batch_size))
```
Comment (apeforest) [inline examples/mxnet_imagenet_resnet50.py]: nit: do you need to cast to `int` for the floor division? Also, maybe add `from __future__ import division` to make it python2 and python3 compatible?

Comment (yuxihu) [reply examples/mxnet_imagenet_resnet50.py]: Did not think about it. Actually I did not change the formula. I think Carl also used the same [one](https://github.com/aws-samples/deep-learning-models/blob/master/models/resnet/mxnet/train_imagenet.py#L315) from Rahul.

**Comment by apeforest** [comment]
maybe you also want to remove `class LogValidationMetricsCallback(object):` since it's no longer used in this example.

**Comment by yuxihu** [comment]
Removed LogValidationMetricsCallback

**Pull request merged**